libyul/optimiser/UnusedFunctionParameterPruner.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * UnusedFunctionParameterPruner: Optimiser step that removes unused parameters from function
+ * definition.
+ */
+
+#include <libyul/optimiser/UnusedFunctionParameterPruner.h>
+#include <libyul/optimiser/UnusedFunctionsCommon.h>
+#include <libyul/optimiser/OptimiserStep.h>
+#include <libyul/optimiser/NameCollector.h>
+#include <libyul/optimiser/NameDisplacer.h>
+#include <libyul/optimiser/NameDispenser.h>
+#include <libyul/YulString.h>
+#include <libyul/AsmData.h>
+#include <libyul/Dialect.h>
+
+#include <libsolutil/CommonData.h>
+
+#include <algorithm>
+#include <optional>
+#include <variant>
+
+using namespace std;
+using namespace solidity::util;
+using namespace solidity::yul;
+
+namespace
+{
+
+bool anyFalse(vector<bool> const& _mask)
+{
+	return any_of(_mask.begin(), _mask.end(), [](bool b){ return !b; });
+}
+
+template <typename K, typename V>
+optional<V> optionalFind(map<K, V> const& _map, K const& _key)
+{
+	auto it = _map.find(_key);
+
+	if (it != _map.end())
+		return it->second;
+
+	return nullopt;
+}
+
+
+} // anonymous namespace
+
+void UnusedFunctionParameterPruner::run(OptimiserStepContext& _context, Block& _block)
```

Comment:
```suggestion
void UnusedFunctionParameterPruner::run(OptimiserStepContext& _context, Block& _ast)
```
(because it is not meant to be just "some" block)